File: Hashing/RadixConversionHashing.java
// File Name: RadixConversionHashing.java

public class RadixConversionHashing {

    static final int TABLE_SIZE = 10;
    static final int BASE = 13;

    static long convertToDecimal(long key) {

        long decimal = 0;
        int power = 0;

        while (key > 0) {
            int digit = (int)(key % 10);
            decimal += digit * Math.pow(BASE, power);
            key = key / 10;
            power++;
        }

        return decimal;
    }

    static int hashFunction(long key) {
        long decimalValue = convertToDecimal(key);
        return (int)(decimalValue % TABLE_SIZE);
    }

    public static void main(String[] args) {

        long[] keys = {210485L, 12345L, 45678L};

        System.out.println("Radix Conversion Method Hashing");
        System.out.println("Table Size = " + TABLE_SIZE + "\n");

        for (long key : keys) {
            int index = hashFunction(key);
            System.out.println("Key (base " + BASE + "): " +
                               key + " -> Hash Index: " + index);
        }
    }
}

File: Hashing/radix_conversion_hashing.py
# File Name: radix_conversion_hashing.py

TABLE_SIZE = 10
BASE = 13

def convert_to_decimal(key):
    decimal = 0
    power = 0

    while key > 0:
        digit = key % 10
        decimal += digit * (BASE ** power)
        key //= 10
        power += 1

    return decimal

def hash_function(key):
    decimal_value = convert_to_decimal(key)
    return decimal_value % TABLE_SIZE

def main():
    keys = [210485, 12345, 45678]

    print("Radix Conversion Method Hashing")
    print("Table Size =", TABLE_SIZE, "\n")

    for key in keys:
        index = hash_function(key)
        print(f"Key (base {BASE}): {key} -> Hash Index: {index}")

if __name__ == "__main__":
    main()
